samples/buildstrategy/buildah/buildstrategy_buildah_cr.yaml

```diff
@@ -149,10 +149,18 @@ spec:
             "${image}" \
             "docker://${image}"
 
-          # Store the digest result
-          buildah images \
-            --format='{{.Digest}}' \
-            "${image}" | tr -d "\n" > '$(results.shp-image-digest.path)'
+          # Store the digest result. This is more complex than expected. BuildAh locally calculates a wrong digest.
+          # We therefore tag the image to a dummy name so that the layers are still present. Then we remove the local
+          # tag. Then we pull again. Then the local digest is correct.
+          # This should be validated again with a newer BuildAh version.
+          # https://github.com/containers/buildah/issues/3866
```

Comment:
This is an interesting topic. From the issue there might be a design difference between buildah and docker(_therefore we think this might be a wrong digest_). If by design Buildah local build digest != push to registry digest, then I think we should not modify this. However, if this is not by design and there is a bug, then yes. By reading the issue, I was not able to understand what is the root cause. I would like to discuss this via a different issue, and hold on this Buildah changes. Might I ask, how did you find this?

Reply:
I to be honest do not know what BuildAh was doing there. For me, the digest is always coming from the registry. Therefore it is normal for me that after you do a `docker build`, that the image locally has no digest (`docker images --digests`). Once you pushed it, there is a digest.

I was not aware that BuildAh tried to calculate a local digest until I added the test case. I have no idea why they are trying to do this. Or, I would be okay if they do it, if they would update it with `buildah push` to the one the registry was calculating.

I googled which is how I found the issue.

Scenario-wise, the locally calculated digest does not have any value to Shipwright users because they don't use the image from inside the BuildAh container. They consume the resulting image from the registry and need to know this digest.

Based on that, I want to include this change. We should definately not go back to the solution with the wrong digest. One could change the strategy to not store any digest at all, but I strongly prefer to set the right one even if the code looks weird for the time being until either BuildAh fixes this, or until the strategy gets rewritten for the shipwright-managed push.

Reply:
thanks @SaschaSchwarze0 that helps. Can you open an issue to follow on this? main reason on adding an extra `pull`,if we can remove this eventually, we should. With the issue we can follow on what @gabemontero finds out.

Reply:
https://github.com/shipwright-io/build/issues/1028